enterprise/server/raft/replica/replica.go

```diff
@@ -705,6 +705,51 @@ func (sm *Replica) loadRangeLease(db ReplicaReader) {
 	sm.setRangeLease(constants.LocalRangeLeaseKey, buf)
 }
 
+// clearInMemoryReplicaState clears in-memory replica state.
+func (sm *Replica) clearInMemoryReplicaState() {
+	sm.rangeMu.Lock()
+	sm.rangeDescriptor = nil
```

Comment:
if a replica is opened and then raft calls RestoreFromSnapshot -- will clearReplicaState call store.Remove() before the replica is rewritten from the snapshot?

Reply:
I added store.RemoveRange() in ClearReplicaState if sm.rangeDescriptor is not nil